query-engine/connectors/sql-query-connector/src/query_builder/select.rs

```diff
@@ -311,13 +396,41 @@ impl<'a> SelectBuilderExt<'a> for Select<'a> {
                 }
                 _ => acc,
             })
+            .with_virtuals_from_selection(selected_fields)
+    }
+
+    fn with_virtuals_from_selection(self, selected_fields: &FieldSelection) -> Select<'a> {
+        build_virtual_selection(selected_fields.virtuals())
+            .into_iter()
+            .fold(self, |select, (alias, expr)| select.value(expr.alias(alias)))
     }
 
     fn with_columns(self, columns: ColumnIterator) -> Select<'a> {
         columns.into_iter().fold(self, |select, col| select.column(col))
     }
 }
 
+fn build_join_conditions(
+    left_columns: impl Iterator<Item = Column<'static>>,
+    right_columns: impl Iterator<Item = Column<'static>>,
```

Comment:
`right_columns` is always `ColumnIterator` in practice but `left_columns` is a `ColumnIterator` in half of the cases and `std::vec::IntoIter` in another half of the cases. We could always require `ColumnIterator` here, then we'll have one less monomorphized instance of this function in the compiled binary but a few more allocation as the `ColumnIterator` boxes the inner iterator.

Reply:
Do we care for size?